dandiapi/api/scheduled_tasks.py

```diff
@@ -3,30 +3,40 @@
 
 This module is imported from celery.py in a post-app-load hook.
 """
+from datetime import timedelta
 
 from celery import shared_task
-from celery.schedules import crontab
 from celery.utils.log import get_task_logger
+from django.conf import settings
 from django.db.transaction import atomic
 
 from dandiapi.api.models import Version
-from dandiapi.api.tasks import write_manifest_files
+from dandiapi.api.tasks import validate_version_metadata, write_manifest_files
 
 logger = get_task_logger(__name__)
 
 
 @shared_task
 @atomic
-def write_draft_manifest_files():
-    logger.info('Writing manifest files for all draft versions')
-    # TODO Optimize this if it causes too much load in production.
-    # Rewriting every draft manifest every time is guaranteed not to miss any changes,
-    # so just do that for now to avoid the complexity involved with modification dates.
-    for draft_version in Version.objects.filter(version='draft').all():
+def validate_draft_version_metadata():
+    logger.info('Checking for draft versions that need validation')
+    # Select only the id of draft versions that have status PENDING
+    pending_draft_versions = (
+        Version.objects.filter(status=Version.Status.PENDING).filter(version='draft').values('id')
+    )
+    logger.info('Found %s versions to validate', pending_draft_versions.count())
```

Comment:
It might be good to list out the IDs found as well, perhaps at a more verbose log level.

Reply:
They will be logged when the task is picked up off the queue, I don't think it's that useful to log them here.